### DISCUSSION TITLE
How are data injected into `org.jberet.test.infinispanRepository.xml`?

### DISCUSSION THREAD

**Discussion by liweinan** [jberet/jsr352#365] [Q&A] [answered]
@chengfang I have a question about this file:

- https://github.com/jberet/jsr352/blob/main/test-apps/common/src/main/resources/META-INF/batch-jobs/org.jberet.test.infinispanRepository.xml

How are the data loaded during the execution of:

- https://github.com/jberet/jsr352/blob/main/test-apps/infinispanRepositoryJdbc/src/test/java/org/jberet/testapps/infinispanRepositoryJdbc/JdbcInfinispanRepositoryIT.java

It seems I can't find where the data is loaded into the tables:

- https://github.com/jberet/jsr352/blob/main/test-apps/infinispanRepositoryJdbc/src/main/resources/jdbc-infinispan.xml

I'm asking because this is related with: https://github.com/jberet/jsr352/pull/364

**Comment by chengfang** [comment, marked as answer]
https://github.com/jberet/jsr352/blob/main/test-apps/infinispanRepositoryJdbc/src/main/resources/jberet.properties specifies that for this test app, infinispan job repository will be used. It also specifies the infinispan config file to use.

So for `JdbcInfinispanRepositoryIT`, `jdbc-infinispan.xml` is the infinispan config file, which configures infinispan to use in-memory h2 database connection.

**Comment by liweinan** [comment]
@chengfang Thanks for teaching!

**Comment by liweinan** [comment]
@chengfang In addition, I tried to check the input data by exporting the database to a file:

https://github.com/jberet/jsr352/compare/main...liweinan:h2_file?expand=1

And I can see the generated tables and its data by loading the database in IntelliJ:

<img width="1432" alt="image" src="https://github.com/jberet/jsr352/assets/201907/4fd28036-a7d7-4cf7-9fc7-a7d07fd1f9a7">

Where are the sources of these data in the table? It seems I can't find the data input source anywhere.

**Comment by chengfang** [comment]
I think `DATA_COLUMN` is a serialized form of the job instance data created by infinispan. The jdbc store used by infinispan does not map to the table used in the jdbc job repository.

See https://github.com/jberet/jsr352/blob/main/jberet-core/src/main/java/org/jberet/repository/InfinispanRepository.java#L317

**Reply by liweinan** [reply]
Now I understand, thanks!
